src/ui/Templates/TemplateFieldsEvaluator.ts

```diff
@@ -1,23 +1,41 @@
 import { IQueryResult } from '../../rest/QueryResult';
 import { IFieldsToMatch } from './Template';
-import * as _ from 'underscore';
+import { each, isArray } from 'underscore';
 
 export class TemplateFieldsEvaluator {
   public static evaluateFieldsToMatch(toMatches: IFieldsToMatch[], result: IQueryResult): boolean {
     let templateShouldBeLoaded = true;
-    _.each(toMatches, (toMatch: IFieldsToMatch) => {
+    each(toMatches, (toMatch: IFieldsToMatch) => {
       let matchAtLeastOnce = false;
       if (!toMatch.values) {
         matchAtLeastOnce = result.raw[toMatch.field] != null;
       } else {
-        _.each(toMatch.values, value => {
+        each(toMatch.values, value => {
           if (!matchAtLeastOnce) {
-            matchAtLeastOnce = result.raw[toMatch.field] && result.raw[toMatch.field].toLowerCase() == value.toLowerCase();
+            const fieldValue: string | string[] = result.raw[toMatch.field];
+            const fieldValues = TemplateFieldsEvaluator.getFieldValueAsArray(fieldValue);
+            matchAtLeastOnce = TemplateFieldsEvaluator.isMatch(fieldValues, value);
           }
         });
       }
       templateShouldBeLoaded = templateShouldBeLoaded && matchAtLeastOnce;
     });
     return templateShouldBeLoaded;
   }
+
+  private static getFieldValueAsArray(fieldValue: string | string[]): string[] {
+    if (typeof fieldValue === 'string') {
+      return [fieldValue];
+    }
+
+    if (isArray(fieldValue)) {
+      return fieldValue;
+    }
+
+    return [];
```

Comment:
Should there be some kind of logging happening here if we somehow end up here ?

Reply:
Or, if it's relatively safe to assume that fieldValue is not something else than a string or string[], remove the first if condition and return return [fieldValue], or [`${fieldvalue}`]

Reply:
Hmm, I am not sure which is better. I was trying to be defensive so that if one day the index/search-api sends a new type (e.g. boolean), the field condition would fail gracefully. At the same time, we already have a try/catch which uses the Default template as the last resort and logs an error.

I think the best will be to simplify as JF suggested.

Thanks for your feedback,